models/metrics/metrics.go

```diff
@@ -0,0 +1,140 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/Azure/arn-sdk/models/v3/schema/types"
+)
+
+const (
+	subsystem    = "arn"
+	successLabel = "success"
+	inlineLabel  = "inline"
+	timeoutLabel = "timeout"
+)
+
+var (
+	eventSentCount      metric.Int64Counter
+	eventSentBytesCount metric.Int64Counter
+	eventSentLatency    metric.Int64Histogram
+
+	currentPromiseCount metric.Int64UpDownCounter
+	promiseCount        metric.Int64Counter
+	promiseLatency      metric.Int64Histogram
+)
+
+func metricName(name string) string {
+	return fmt.Sprintf("%s_%s", subsystem, name)
+}
+
+// Init initializes the arn model metrics. This should only be called by the tattler constructor or tests.
+func Init(meter metric.Meter) error {
+	var err error
+	eventSentCount, err = meter.Int64Counter(metricName("event_sent_total"), metric.WithDescription("total number of events sent by the ARN client"))
+	if err != nil {
+		return err
+	}
+
+	eventSentBytesCount, err = meter.Int64Counter(metricName("event_sent_bytes_total"), metric.WithDescription("total number of bytes in event data sent by the ARN client"))
+	if err != nil {
+		return err
+	}
+
+	// TODO: adjust buckets
+	eventSentLatency, err = meter.Int64Histogram(
+		metricName("event_sent_ms"),
+		metric.WithDescription("time spent to send ARN event"),
+		metric.WithExplicitBucketBoundaries(50, 100, 200, 400, 600, 800, 1000, 1250, 1500, 2000, 3000, 4000, 5000, 10000, 60000, 300000, 600000),
+	)
+	if err != nil {
+		return err
+	}
+
+	promiseCount, err = meter.Int64Counter(metricName("promise_total"), metric.WithDescription("total number of promises made by the ARN client"))
```

Comment:
Do you think we need these?

We are getting counts of messages being sent.  If all messages have promises, then we have the same count.  currentPromiseCount is interesting, because this means how many outstanding calls we are waiting on.  But I'm not sure promiseCount is buying us anything.

Reply:
I added it for the timeouts mostly, so we can count how many time out. Since it's awkward to call the event sent failure function when getting ErrPromiseTimeout in Promise()

Reply:
Sounds good.